react/week5/githubusersprofile/src/Components/Profile.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import "./Profile.css";
+
+export function Profile({ user }) {
+  const [loadedUser, setLoadedUser] = useState(user);
+  const { userName } = useParams();
+
+  //useState is for tracking user's properties and their changes.
+  //useParams returns an object of key/value pairs of URL parameters, in ourcase is "userName"
+  //param is loaded inside profile component, hence its imported in this component.
+  //useEffect is for fetching.
+
+  useEffect(() => {
+    if (user && user.id) return; //if there is user and has user.id, then fetch his/her details from github API.
+
+    fetch(`https://api.github.com/users/${userName}`)
+      .then((res) => res.json())
+      .then((u) => setLoadedUser(u))
+      .catch((e) => console.error("Error fetching user", e));
+  }, [user, userName]);
+
+  if (!loadedUser) return "Loading..."; //if the user is not found, return Loading
+
+  const {
+    name,
+    html_url,
+    login,
+    avatar_url,
+    company,
+    followers,
+    following,
+    public_repos,
+  } = loadedUser; //these are user's object properties as per githubs data dashboard.
+  return (
+    <div className="profile">
+      <div>
+        <img src={avatar_url} alt="" />
```

Comment:
```suggestion
        <img src={avatar_url} alt="avatar picture" />
```
always write alt as description